test/jdk/java/util/zip/DeInflate.java

```diff
@@ -137,14 +136,14 @@ static void checkByteBufferReadonly(Deflater def, Inflater inf,
     static void check(Deflater def, byte[] in, int len, boolean nowrap)
         throws Throwable
     {
-        byte[] tempBuffer = new byte[len];
+        byte[] tempBuffer = new byte[1024];
         byte[] out1, out2;
         int m = 0, n = 0;
         Inflater inf = new Inflater(nowrap);
         def.setInput(in, 0, len);
         def.finish();
 
-        try (ByteArrayOutputStream baos = new ByteArrayOutputStream(len)) {
+        try (ByteArrayOutputStream baos = new ByteArrayOutputStream()) {
```

Comment:
I don't think this is really required because the `ByteArrayOutputStream` will grow dynamically anyway, even if initialized to zero but for all other cases `len` is probably a better estimation than the default size of `32`. But I leave it up to you if you want to change it back.

Reply:
Hello Volker, you are right that even initializing this with `len` should be fine since the `ByteArrayOutputStream` will grow if/when needed. However, the reason I decided to remove the usage of `len` here was to avoid any kind of confusion and make it easier to read that `len` is only for the input length (as noted in the javadoc of that method) and plays no other role in deciding sizes of (temporary) outputs.